File: samplejava/src/main/java/pyxis/uzuki/live/richutils/samplejava/MainActivity.java
package pyxis.uzuki.live.richutils.samplejava;

import android.annotation.SuppressLint;
import android.content.Intent;
import android.graphics.Color;
import android.location.Location;
import android.net.Uri;
import android.os.Bundle;
import android.text.TextUtils;
import android.widget.Button;
import android.widget.ImageView;
import android.widget.TextView;

import org.json.JSONObject;

import java.io.InputStream;
import java.net.URL;

import javax.net.ssl.HttpsURLConnection;

import butterknife.BindView;
import kotlin.Unit;
import pyxis.uzuki.live.richutilskt.utils.ContactItem;
import pyxis.uzuki.live.richutilskt.utils.RichUtils;

public class MainActivity extends BaseActivity {
    @BindView(R.id.logo) ImageView logo;
    @BindView(R.id.txtVersion) TextView txtVersion;
    @BindView(R.id.alert) Button alert;
    @BindView(R.id.bitmap) Button bitmap;
    @BindView(R.id.date) Button date;
    @BindView(R.id.json) Button json;
    @BindView(R.id.permission) Button permission;
    @BindView(R.id.preference) Button preference;
    @BindView(R.id.media) Button media;
    @BindView(R.id.btnReboot) Button btnReboot;
    @BindView(R.id.contact) Button contact;
    @BindView(R.id.location) Button location;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(RichUtils.setTypeface(this, R.layout.activity_main));
        RichUtils.setStatusNavBarColor(this, Color.parseColor("#303F9F"));


        RichUtils.runAsync(() -> {
            getLatestReleaseFromGitHub();
            return Unit.INSTANCE;
        }, 5);

        logo.setOnClickListener(view -> {
            Intent intent = new Intent(Intent.ACTION_VIEW, Uri.parse("https://github.com/WindSekirun/RichUtilsKt"));
            startActivity(intent);
        });

        btnReboot.setOnClickListener(view -> RichUtils.reboot(this));
        alert.setOnClickListener(view -> startActivity(new Intent(MainActivity.this, AlertActivity.class)));
        bitmap.setOnClickListener(v -> startActivity(new Intent(MainActivity.this, BitmapActivity.class)));
        date.setOnClickListener(v -> startActivity(new Intent(MainActivity.this, DateActivity.class)));
        json.setOnClickListener(v -> startActivity(new Intent(MainActivity.this, JSONActivity.class)));
        permission.setOnClickListener(v -> startActivity(new Intent(MainActivity.this, PermissionActivity.class)));
        preference.setOnClickListener(v -> startActivity(new Intent(MainActivity.this, PreferenceActivity.class)));
        media.setOnClickListener(v -> startActivity(new Intent(MainActivity.this, PickMediaActivity.class)));
        contact.setOnClickListener(v -> startActivity(new Intent(MainActivity.this, ContactActivity.class)));
        location.setOnClickListener(v -> startActivity(new Intent(MainActivity.this, LocationActivity.class)));
    }

    private void getLatestReleaseFromGitHub() {
        String apiPath = "https://api.github.com/repos/windsekirun/richutilskt/releases/latest";
        try {
            URL url = new URL(apiPath);
            HttpsURLConnection connection = (HttpsURLConnection) url.openConnection();
            connection.setConnectTimeout(10000);
            connection.setRequestMethod("GET");

            int resCode = connection.getResponseCode();
            if (resCode == HttpsURLConnection.HTTP_OK) {
                InputStream stream = connection.getInputStream();
                String response = RichUtils.getString(stream);
                runOnUiThread(() -> parsingReleaseData(response));
                connection.disconnect();
            }
        } catch (Exception e) {
            e.printStackTrace();
        }
    }

    @SuppressLint("SetTextI18n")
    private void parsingReleaseData(String response) {
        if (TextUtils.isEmpty(response))
            return;

        JSONObject jsonObject = RichUtils.createJSONObject(response);
        if (jsonObject != null) {
            String version = RichUtils.getJSONString(jsonObject, "tag_name", "");
            txtVersion.setText("Latest version -> " + version);
        }
    }
}

File: sample/src/main/java/pyxis/uzuki/live/richutilssample/MainActivity.kt
package pyxis.uzuki.live.richutilssample

import android.annotation.SuppressLint
import android.graphics.Color
import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.text.TextUtils
import android.util.Log
import kotlinx.android.synthetic.main.activity_main.*
import org.jetbrains.anko.browse
import org.jetbrains.anko.startActivity
import org.json.JSONObject
import pyxis.uzuki.live.richutilskt.utils.*
import java.net.HttpURLConnection
import java.net.URL

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(setTypeface(R.layout.activity_main))

        setStatusNavBarColor(Color.parseColor("#303F9F"))

        Log.d("goat", userManager.isUserAGoat.toString() + "")

        runAsync { getLatestReleaseFromGitHub() }

        alert.setOnClickListener { startActivity<AlertActivity>() }
        bitmap.setOnClickListener { startActivity<BitmapActivity>() }
        date.setOnClickListener { startActivity<DateActivity>() }
        media.setOnClickListener { startActivity<PickMediaActivity>() }
        btnReboot.setOnClickListener { this.reboot() }
        location.setOnClickListener { startActivity<LocationActivity>() }
        logo.setOnClickListener { browse("https://github.com/WindSekirun/RichUtilsKt") }
        json.setOnClickListener { startActivity<JSONActivity>() }
        permission.setOnClickListener { startActivity<PermissionActivity>() }
        preference.setOnClickListener { startActivity<PreferenceActivity>() }
        contact.setOnClickListener { startActivity<ContactActivity>() }
        misc.setOnClickListener { startActivity<MiscActivity>() }
        inapp.setOnClickListener { startActivity<InAppActivity>() }
    }

    private fun getLatestReleaseFromGitHub() {
        val apiPath = "https://api.github.com/repos/windsekirun/richutilskt/releases/latest"

        val url = URL(apiPath)
        val conn = url.openConnection() as HttpURLConnection

        val resCode = conn.responseCode
        if (resCode == HttpURLConnection.HTTP_OK) {
            val inputStream = conn.inputStream
            val response: String = inputStream.getString()
            runOnUiThread { parsingReleaseData(response) }
        }

        conn.disconnect()
    }

    @SuppressLint("SetTextI18n")
    private fun parsingReleaseData(response: String) {
        if (TextUtils.isEmpty(response))
            return

        val jsonObject: JSONObject? = response.createJSONObject()
        val version = jsonObject?.getJSONString(name = "tag_name")

        txtVersion.setHtmlText(String.format(getString(R.string.txt_version), version))
    }
}

